### DISCUSSION TITLE
Update created project with different answer location

### DISCUSSION THREAD

**Discussion by nilsdebruin** [copier-org/copier#859] [Q&A] [answered]
Hi all!

I ran into the following (as a further iteration on another discussion: https://github.com/copier-org/copier/discussions/714):

I have created a copier template which can be found here:

https://github.com/nilsdebruin/copier-template-report

In this repository, I created a copier.yml with an answer-file settings, like so:

```
_subdirectory: copier-template
_answers_file: .copier-answers-copier-template-4afc07a0-63f3-4e21-af38-3ccbb3e92ffe.yml
```

The template has the following structure for the answer file:

`copier-template/copier-instances/{{ _copier_conf.answers_file }}.jinja`

When I create a new instance of the template with the following command:

`copier gh:nilsdebruin/copier-template-report test`

a new directory is created with the files from this template and the answer file is stored in:

`test/copier-instances/.copier-answers-copier-template-4afc07a0-63f3-4e21-af38-3ccbb3e92ffe.yml`

as I would expect.

If I now go to the test folder and run the following command:

`copier -a ./copier-instances/.copier-answers-copier-template-4afc07a0-63f3-4e21-af38-3ccbb3e92ffe.yml update`

I get the following error message:

`FileNotFoundError: [Errno 2] No such file or directory: '/var/folders/3_/ly7vn0c50l9c84dv4p82yfzw0000gn/T/copier.main.update_diff.g8os_1_e/copier-instances/copier-instances/.copier-answers-copier-template-4afc07a0-63f3-4e21-af38-3ccbb3e92ffe.yml'`

If I move the answer file to the root of the test folder and run the following command it works:

`copier -a .copier-answers-copier-template-4afc07a0-63f3-4e21-af38-3ccbb3e92ffe.yml update`

I would like to be able to store the answer files in a sub-directory of the created test folder.

Is this possible and if not, what would be needed to make this possible?

**Comment by yajo** [comment, marked as answer]
I'd love that to work if it doesn't. This way, we could follow https://dot-config.github.io/ standard. However, if it doesn't work, then somebody has to do it. Would you like to volunteer on that?

If so, just make sure to add test and docs to the PR, so the feature gets stable just like almost any other Copier feature.

**Reply by nilsdebruin** [reply]
Great, I did not know of this standard! Thanks for pointing this out; this is really nice! Is there a document / link I can follow to run the tests / build when I want to give it a shot? I would really love to help and don't want to take too much of your time. Thanks in advance and also for creating this library, it has great potential!

**Reply by yajo** [reply]
I'd start with https://copier.readthedocs.io/en/latest/contributing/

**Reply by nilsdebruin** [reply]
Based on the discussion, I have created a draft pull request here:

https://github.com/copier-org/copier/pull/869

This is a first shot at the issue, so it could be that you want a different direction to solve this. I already learned a lot from debugging the code, so that is already a nice learning for me!

**Reply by nilsdebruin** [reply]
I created a new pull request, which includes the updates which have been made to master for rejection, see: https://github.com/copier-org/copier/pull/870

**Reply by nilsdebruin** [reply]
Hi @yajo let me know what I can do to improve the pull request!

**Reply by nilsdebruin** [reply]
@yajo I removed the pull request and will re-think how I can approach it again!

**Reply by nilsdebruin** [reply]
I have now first created a test which indicates what is not working currently, is this a better start to approach the problem?

https://github.com/copier-org/copier/pull/927

Let me know!

**Reply by nilsdebruin** [reply]
During debugging the code, I can see that in the function: ```def _render_path``` the path gets extended with an extra part.

This results in a rendered path list of: ```['.config', '.config/altered-answers.yml.jinja']```

In the end this results in the fact that the path cannot be resolved, as the answers file is looked for in: ```.config/.config/altered-answers.yml```

Where do you think would be the best place to fix this in the code? Thanks in advance and thanks for your time!

![Screenshot 2023-01-25 at 11 00 23](https://user-images.githubusercontent.com/4271579/214534424-0bb83676-b6fb-46c3-9fe5-661b2e34a55b.png)

**Reply by nilsdebruin** [reply]
I took a first shot at it, hopefully, it makes sense, the pull request has been updated!

**Reply by nilsdebruin** [reply]
@yajo thanks for all the effort, great that the changes could be merged! I will mark this discussion as answered!

**Reply by yajo** [reply]
Thank you for your patience! You did the hard work. 😊
